src/app/clusters/resource-monitoring-server/resource-monitoring-server.h

```diff
@@ -0,0 +1,161 @@
+    bool mInPlaceIndicator                         = true;
+    DataModel::Nullable<uint32_t> mLastChangedTime;
+
+    uint32_t mFeatureMap;
+
+    bool mResetConditionCommandSupported = false;
+
+    /**
+     * This checks if the clusters instance is a valid ResourceMonitoring cluster based on the AliasedClusters list.
+     * @return true     if the cluster is a valid ResourceMonitoring cluster.
+     */
+    bool IsValidAliasCluster() const;
+
+    /**
+     * Internal method to handle the ResetCondition command.
+     */
+    void HandleResetCondition(HandlerContext & ctx,
+                              const ResourceMonitoring::Commands::ResetCondition::DecodableType & commandData);
+
+public:
+    /**
+     * Creates a resource monitoring cluster instance. The Init() method needs to be called for this instance to be registered and
+     * called by the interaction model at the appropriate times.
+     * @param aEndpointId   The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId    The ID of the ResourceMonitoring aliased cluster to be instantiated.
+     * @param aDelegate     A pointer to a delegate that will handle application layer logic.
+     */
+    Instance(EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeatureMap,
+             ResourceMonitoring::Attributes::DegradationDirection::TypeInfo::Type aDegradationDirection,
+             bool aResetConditionCommandSupported) :
+        CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mDegradationDirection(aDegradationDirection),
+        mResetConditionCommandSupported(aResetConditionCommandSupported)
+    {
+        mEndpointId = aEndpointId;
+        mClusterId  = aClusterId;
+        mFeatureMap = aFeatureMap;
+    }
+
+    ~Instance() = default;
+    // Not copyable or movable
+    Instance(const Instance &) = delete;
+    Instance & operator=(const Instance &) = delete;
+
+    template <typename RequestT, typename FuncT>
+    void HandleCommand(HandlerContext & handlerContext, FuncT func);
+
+    // The following methods should be overridden by the SDK user to implement the business logic of their application
+
+    /**
+     * This init method will be called during Resource Monitoring Server initialization after the instance information has been
+     * validated and the instance has been registered. This method should be overridden by the SDK user to initialize the
+     * application logic.
+     *
+     * @return CHIP_NO_ERROR    If the application was initialized successfully. All other values will cause the initialization to
+     * fail.
+     */
+    virtual CHIP_ERROR AppInit() = 0;
+
+    /**
+     * This method is to be overridden by a user implemented method to handle the application specifics of the ResetCondition
+     * command.
```

Comment:
This should clearly document which parts the cluster impl will and will not handle itself (especially of the set of things the spec requires), so it's possible to implement OnResetCondition without reading the code of the cluster.

Again, ideally we would do all the spec-required bits in the cluster itself, but followup is fine for that.

Reply:
Updated the comment to clarify that it's all up to the sdk user to implement the resets